tensorboard/pip_package/pip_smoke_test.sh

```diff
@@ -131,32 +131,51 @@ TMP_LOGDIR=$(mktemp -d --suffix _tensorboard_logdir)
 tensorboard --port="${TEST_PORT}" --logdir="${TMP_LOGDIR}" &
 TB_PID=$!
 
-TEST_URL="http://localhost:${TEST_PORT}/data/logdir"
 echo
 echo "tensorboard binary should be running at pid ${TB_PID}"
-echo "Sending test HTTP requests at URL: ${TEST_URL} (${NUM_RETRIES} retries)"
-echo
-
-RETRY_COUNTER=0
-REQUEST_SUCCEEDED=0
-while [[ "${RETRY_COUNTER}" -lt "${NUM_RETRIES}" ]]; do
-  sleep 1
-  STATUS_CODE=$(curl -Is "${TEST_URL}" | head -1 | cut -d ' ' -f 2)
-  if [[ "${STATUS_CODE}" == 200 ]]; then
-    echo
-    echo "Request to ${TEST_URL} succeeded (200)!"
-    echo
-    REQUEST_SUCCEEDED=1
-    break
-  else
-    : $(( RETRY_COUNTER++ ))
-    echo "Request to ${TEST_URL} failed. Will retry in 1 second..."
+echo
+
+function test_access_url(){
+  # Test requests to a URL with curl.
+  #
+  # Usage:
+  #   test_access_url <URL>
+  # E.g.,
+  #   test_access_url http://localhost:6006/
+  TEST_URL="$1"
+
+  echo
+  echo "Sending test HTTP requests at URL: ${TEST_URL} (${NUM_RETRIES} retries)"
+  echo
+
+  RETRY_COUNTER=0
+  REQUEST_SUCCEEDED=0
+  while [[ "${RETRY_COUNTER}" -lt "${NUM_RETRIES}" ]]; do
+    sleep 1
+    STATUS_CODE=$(curl -Is "${TEST_URL}" | head -1 | cut -d ' ' -f 2)
+    if [[ "${STATUS_CODE}" == 200 ]]; then
+      echo
+      echo "Request to ${TEST_URL} succeeded (200)!"
+      echo
+      REQUEST_SUCCEEDED=1
```

Comment:
Can simplify this state by just `return`ing here, removing this variable, and dissolving the if-statement on line 168.